src/providers/AppProviders.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { createContext, useContext, useState } from "react";
+
+type UserContextType = {
+  firstName?: string;
+  lastName?: string;
+  isAuth: boolean;
+  email?: string;
+  setFirstName?: (firstName: string) => void;
+  setLastName?: (lastName: string) => void;
+  setIsAuth: (isAuth: boolean) => void;
+  setEmail?: (email: string) => void;
+};
+
+const userContextDefaultValues: UserContextType = {
+  firstName: "",
+  lastName: "",
+  isAuth: false,
+  email: "",
+  setFirstName: (firstName: string) => {},
+  setLastName: (lastName: string) => {},
+  setIsAuth: (isAuth: boolean) => {},
+  setEmail: (email: string) => {},
+};
+
+export const UserContext = createContext<UserContextType>(
+  userContextDefaultValues
+);
+
+export const AppProviders = ({ children }: { children: React.ReactNode }) => {
```

Comment:
Wydaje mnie się, że to raczej jest `UserContextProvider`

Reply:
I można wrzucić wtedy w `src/contexts`.
Jeśli chciałabyś dodać `AppProviders`, to myślę, że on może wykorzystywać w sobie `UserContextProvider`.